pydicom/tests/test_framereader.py

```diff
@@ -0,0 +1,420 @@
+from pydicom.filereader import data_element_offset_to_value
+from pydicom.pixel_data_handlers import gdcm_handler
+from pydicom.tag import ItemDelimiterTag
+
+
+have_numpy = config.have_numpy
+if have_numpy:
+    import numpy  # NOQA
+
+have_gdcm_handler = gdcm_handler.is_available()
+try:
+    from PIL import Image as PILImg
+except ImportError:
+    # If that failed, try the alternate import syntax for PIL.
+    try:
+        import Image as PILImg
+    except ImportError:
+        # Neither worked, so it's likely not installed.
+        PILImg = None
+have_pillow = PILImg is not None
+
+can_decode = all([have_gdcm_handler, have_pillow, have_numpy])
+
+encaps_paths = [
+    get_testdata_file(name)
+    for name in [
+        "SC_rgb_jpeg.dcm",
+        "SC_rgb_dcmtk_+eb+cy+s2.dcm",
+        "JPGExtended.dcm",
+        "SC_rgb_dcmtk_+eb+cr.dcm",
+        "MR_small_RLE.dcm",
+        "rtdose_rle_1frame.dcm",
+        "SC_rgb_dcmtk_+eb+cy+s4.dcm",
+        "SC_rgb_rle.dcm",
+        "MR_small_jp2klossless.dcm",
+        "693_J2KI.dcm",
+        "SC_rgb_jpeg_app14_dcmd.dcm",
+        "SC_rgb_rle_2frame.dcm",
+        "SC_rgb_dcmtk_+eb+cy+n1.dcm",
+        "SC_rgb_dcmtk_+eb+cy+np.dcm",
+        "SC_jpeg_no_color_transform_2.dcm",
+        "SC_rgb_rle_16bit_2frame.dcm",
+        "SC_rgb_dcmtk_+eb+cy+n2.dcm",
+        "SC_rgb_jpeg_dcmtk.dcm",
+        "SC_rgb_jpeg_lossy_gdcm.dcm",
+        "SC_jpeg_no_color_transform.dcm",
+        "SC_rgb_small_odd_jpeg.dcm",
+        "GDCMJ2K_TextGBR.dcm",
+        "SC_rgb_gdcm_KY.dcm",
+        "JPEG2000-embedded-sequence-delimiter.dcm",
+        "SC_rgb_rle_32bit_2frame.dcm",
+        "SC_rgb_jpeg_gdcm.dcm",
+        "JPEG-lossy.dcm",
+        "JPEG2000.dcm",
+        "MR_small_jpeg_ls_lossless.dcm",
+        "SC_rgb_rle_32bit.dcm",
+        "J2K_pixelrep_mismatch.dcm",
+        "rtdose_rle.dcm",
+        "SC_rgb_rle_16bit.dcm",
+    ]
+]
+
+cannot_read = [
+    "rtplan_truncated.dcm",  # missing PixelData
+    "no_meta_group_length.dcm",  # missing PixelData
+    "rtstruct.dcm",  # missing PixelData
+    "reportsi_with_empty_number_tags.dcm",  # missing PixelData
+    "meta_missing_tsyntax.dcm",  # Missing BitsAllocated and other attr
+    "waveform_ecg.dcm",  # missing PixelData
+    "nested_priv_SQ.dcm",  # Missing BitsAllocated and other attr
+    "no_meta.dcm",  # missing PixelData
+    "empty_charset_LEI.dcm",  # missing PixelData
+    "rtplan.dcm",  # missing PixelData
+    "ExplVR_LitEndNoMeta.dcm",  # missing PixelData
+    "UN_sequence.dcm",  # missing PixelData
+    "reportsi.dcm",  # missing PixelData
+    "priv_SQ.dcm",  # missing PixelData
+    "test-SR.dcm",  # missing PixelData
+    "image_dfl.dcm",  # missing PixelData
+    "ExplVR_BigEndNoMeta.dcm",  # missing PixelData
+    "emri_small_jpeg_2k_lossless_too_short.dcm",  # truncated
+]
+
+can_read_paths = [
+    f
+    for f in get_testdata_files("*.dcm")
+    if not any([f.endswith(x) for x in cannot_read])
+]
+
```

Comment:
Easier and more readable, I think, to make `can_read` a `set`, and then `all_paths=(Path(f) for f in get_testdata_files("*.dcm"))`, and `can_read_paths = [p for p in all_paths if p.name not in cannot_read]`.

Really pydicom should have a `get_testdata_paths` function since we've been trying to move more towards pathlib; something for a future PR if we can remember...

Reply:
Changed [here](https://github.com/pydicom/pydicom/blob/6d2c73aafea9529e756de0ab41a4d923d4ea0da3/pydicom/tests/test_framereader.py#L75)